### ISSUE TITLE
[FUTURE] Call for ideas: Form builder

### ISSUE THREAD

**Issue by haringsrob** [area17/twill#1656]
## Summary

It can be useful for site admins to build dynamic forms. I want to explore if we can make this possible in Twill core.

## Describe the solution you'd like

Since we now have the form builder, we could just make a UI in front of it that saves that to a JSON model.

- Introduce a "Form" model and a "FormSubmissions" model for storing the form and its submissions.
- Introduce a pluggable (packageable) action such as "mail" that can be executed when the form is submitted.

When a form is submitted, it is stored in the FormSubmissions, and one or more actions are being executed, examples of actions:

- Send an e-mail
- Send via an API (hubspot, ...)
- Conver to another type of entity

- Introduce a generic block "Form" that has the sole purpose of rendering (coupled) or exposing the form (decoupled). This form can be placed in any page with a block editor.

## Additional context

If you have additional ideas or suggestions, let's all add them as comments and discuss.

**Comment by kallefrombosnia** [comment]
Form custom actions
I would like to make custom actions in the form or sidebar (under the Update/ Publish button).
This can be an independent JS function invoked from Twill components on some change or triggered by a button (needs a new Vue component) that can trigger our custom controller method via XHR request.

Currently, I achieve this by hooking Model events or Twills methods like `beforeSave` or `prepareFieldsBeforeSave` etc.
Is this idea doable with the dynamic forms?

**Comment by haringsrob** [comment]
Hey @kallefrombosnia,

This proposal is more about building forms like this: https://form.twill.io/

But your comment should be doable as well. Did you already propose this in another issue?

**Comment by kallefrombosnia** [comment]
No, it wasn't presented anywhere. I remember talking about this only on Discord with `pbovin`.
Ok, but I was referring to dynamic forms where a new dynamic option could provide us with this feature where we could set the method name which will be used as the endpoint.
